dao/src/main/java/org/thingsboard/server/dao/model/sql/AiModelSettingsEntity.java

```diff
@@ -0,0 +1,110 @@
+import java.util.UUID;
+
+@Getter
+@Setter
+@ToString
+@Entity
+@Table(name = ModelConstants.AI_MODEL_SETTINGS_TABLE_NAME)
+public class AiModelSettingsEntity extends BaseVersionedEntity<AiModelSettings> {
+
+    public static final Map<String, String> COLUMN_MAP = Map.of(
+            "createdTime", "created_time",
+            "provider", "(configuration -> 'providerConfig' ->> 'provider')",
+            "modelId", "(configuration -> 'modelConfig' ->> 'modelId')"
+    );
+
+    public static final Set<String> ALLOWED_SORT_PROPERTIES = Collections.unmodifiableSet(
+            new LinkedHashSet<>(List.of("createdTime", "name", "provider", "modelId"))
+    );
+
+    @Column(name = ModelConstants.AI_MODEL_SETTINGS_TENANT_ID_COLUMN_NAME, nullable = false, columnDefinition = "UUID")
+    private UUID tenantId;
+
+    @Column(name = ModelConstants.AI_MODEL_SETTINGS_NAME_COLUMN_NAME, nullable = false)
+    private String name;
+
+    @Type(JsonBinaryType.class)
+    @Column(name = ModelConstants.AI_MODEL_SETTINGS_CONFIGURATION_COLUMN_NAME, nullable = false, columnDefinition = "JSONB")
+    private AiModel<?> configuration;
+
+    @Column(name = ModelConstants.EXTERNAL_ID_PROPERTY, columnDefinition = "UUID")
+    private UUID externalId;
+
+    public AiModelSettingsEntity() {}
+
+    public AiModelSettingsEntity(AiModelSettings aiModelSettings) {
+        super(aiModelSettings);
+        tenantId = getTenantUuid(aiModelSettings.getTenantId());
+        name = aiModelSettings.getName();
+        configuration = aiModelSettings.getConfiguration();
+        externalId = getUuid(aiModelSettings.getExternalId());
+    }
+
+    @Override
+    public AiModelSettings toData() {
+        var settings = new AiModelSettings(new AiModelSettingsId(id));
+        settings.setCreatedTime(createdTime);
+        settings.setVersion(version);
+        settings.setTenantId(TenantId.fromUUID(tenantId));
+        settings.setName(name);
+        settings.setConfiguration(configuration);
+        settings.setExternalId(getEntityId(externalId, AiModelSettingsId::new));
+        return settings;
+    }
+
+    @Override
+    public final boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null) return false;
+        Class<?> oEffectiveClass = o instanceof HibernateProxy ? ((HibernateProxy) o).getHibernateLazyInitializer().getPersistentClass() : o.getClass();
+        Class<?> thisEffectiveClass = this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass() : this.getClass();
+        if (thisEffectiveClass != oEffectiveClass) return false;
+        AiModelSettingsEntity that = (AiModelSettingsEntity) o;
+        return getId() != null && Objects.equals(getId(), that.getId());
+    }
+
+    @Override
+    public final int hashCode() {
+        return this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass().hashCode() : getClass().hashCode();
```

Comment:
Can you please describe the reason why HibernateProxy is respected here and why we don't do that for all other entity types?